### ISSUE TITLE
Basic Documentation on how to use this in a Rails app

### ISSUE THREAD

**Issue by Kinselan** [lazaronixon/trix-extensions#8]
It would be really helpful to see a few lines of documentation to describe how to use this tool in a Rails app with rich text trix editor.
Without any documentation, it's really not useful even though the plugin looks great.

**Comment by lazaronixon** [comment]
It is basically download repository and extract it to your rails project and add `require("richtext")` to /javascript/packs/application.js

**Comment by r-obeen** [comment]
Hello @lazaronixon,
Thank you for your work on this extension ! Where exactly should we extract the repository in our project ? Anywhere in the repo ? Or somewhere like `/javascript/packs/` ? I can't figure it out but I need your extension so badly !

Thanks

**Comment by lazaronixon** [comment]
You should extract it on root directory

**Comment by r-obeen** [comment]
> You should extract it on root directory

Okay, just did it... But I'm getting this error `Module not found: Error: Can't resolve 'richtext' in '/Users/robeen/dev/tourcrib/app/javascript/packs'` any idea what's wrong ?

Here is my file structure :
```
tourcrib
  ...
  trix-extensions
    app
    assets
    etc...
  ...
```

Thanks for your help

**Comment by r-obeen** [comment]
Okay, I kinda managed to make it work... My only issue now is that when I save, the trix editor strips the `<h2-6>` tags as well as the `style=""` attributes. Even tho I added :

```
Rails.application.config.after_initialize do
   ActionText::ContentHelper.allowed_attributes.add 'style'
   
   ActionText::ContentHelper.allowed_tags.add 'h2'
   ...
end
```

Do you have any idea what's up ? It's so frustrating, I've been trying to figure it out for like 4 hours now... Action text is so unflexible by nature right now... It's sad

**Comment by Merovex** [comment]
Agreed. This needs "how-to" text to be generally useful. I'll see if I can figure it out.

**Comment by iagopuccini** [comment]
@r-obeen 

if you just use it does not work? 
In my case, it worked very well.

```
Rails.application.config.after_initialize do
   ActionText :: ContentHelper.allowed_attributes << 'style'
end
```

**Comment by matt-riemer** [comment]
Hey, I can confirm that this `after_initialize` does not work in Rails 6.1.3.1

If I just do `ActionText::ContentHelper.allowed_attributes << 'style'` in a `config/initializers/action_text.rb` (or wherever) it works.

But if I use the `after_initialize do ... end` it does not work! :)

**Comment by driveton** [comment]
To get this working in _current_ Rails versions (>8x):

1) Add the content of this file: https://github.com/lazaronixon/trix-extensions/blob/master/app/javascript/richtext.js to your app/javascript/application.js
(You don't need this: import Trix from "trix" on top)

<img width="250" alt="Image" src="https://github.com/user-attachments/assets/9d759038-3c1a-44a3-ae14-f2d5f4dee530" />

2) Transform this scss to css using AI or your preffered tool: https://github.com/lazaronixon/trix-extensions/blob/master/assets/stylesheets/richtext.scss and copy the file to you app/assets/stylesheets/anynamesincethisisautoloaded.css

3) Copy images to app/assets/images

<img width="250" alt="Image" src="https://github.com/user-attachments/assets/5dce0ee0-e02f-45d5-91fc-1b583617aff3" />

4) That's it assuming you are using Action Text to get Trix on your app.

<img width="250" alt="Image" src="https://github.com/user-attachments/assets/2bef9474-c61d-45f3-a5f4-cd83a6fd3713" />
